polkadotjs-examples/src/examples/register-address.ts

```diff
@@ -16,12 +16,15 @@ export const registerAddress = async () => {
             if (status.isInBlock) {
                 events.forEach(({ event }) => {
                     const types = event.typeDef;
+console.log('=== event');
                     event.data.forEach((data, index) => {
                         console.log(`pallet: ${event.section} event name: ${event.method}`)
                         console.log("event types", types[index].type + '; event data:' + data.toString());
                     });
                 });
+console.log('*** will unsubscribe');
                 unsubscribe();
+console.log('*** after unsubscribe');
```

Comment:
@jfrank-summit I've got a local creditcoin-node running with --dev and then execute `yarn start`. What I see on the terminal is:

```bash
$ yarn start
yarn run v1.22.17
warning package.json: No license field
$ npx ts-node ./src/index.ts
2022-03-24 10:58:49        API/INIT: RPC methods not decorated: creditcoin_eventsSubscribe, creditcoin_eventsUnsubscribe, creditcoin_getEvents
current status is Ready
current status is {"inBlock":"0x810743458c22b190aac07dcaa9617f8e598e4b64740eac866cbc24551208f93c"}
=== event
pallet: balances event name: Withdraw
event types AccountId32; event data:5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty
pallet: balances event name: Withdraw
event types u128; event data:9993803865385478
=== event
pallet: system event name: ExtrinsicFailed
event types {"_enum":{"Other":"Null","CannotLookup":"Null","BadOrigin":"Null","Module":"{\"index\":\"u8\",\"error\":\"u8\"}","ConsumerRemaining":"Null","NoProviders":"Null","TooManyConsumers":"Null","Token":"SpRuntimeTokenError","Arithmetic":"SpRuntimeArithmeticError"}}; event data:{"module":{"index":5,"error":0}}
pallet: system event name: ExtrinsicFailed
event types {"weight":"u64","class":"FrameSupportWeightsDispatchClass","paysFee":"FrameSupportWeightsPays"}; event data:{"weight":125010000,"class":"Normal","paysFee":"Yes"}
*** will unsubscribe
*** after unsubscribe
```

Two remarks here: 
- There is an `ExtrinsicFailed` event. I'm not sure why but doesn't look good. IIRC Bob's account doesn't have any funds. However I've also tried with Alice and got the same result.
- The `registerAddress` function doesn't seem to exit even after calling unsubscribe. I have the process still active on my terminal. Any idea why that is ?

Reply:
@nathanwhit FTR after a hint from Jeremy the result is now

```bash
$ yarn start
yarn run v1.22.17
warning package.json: No license field
$ npx ts-node ./src/index.ts
2022-03-24 11:24:24        API/INIT: RPC methods not decorated: creditcoin_eventsSubscribe, creditcoin_eventsUnsubscribe, creditcoin_getEvents
current status is Ready
current status is {"inBlock":"0x8e82d7d7c049fb7c1b438893dd52cc760ea8adbafd5883078204ce47f76c0389"}
pallet: balances event name: Withdraw
event types AccountId32 event data: 5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty
pallet: balances event name: Withdraw
event types u128 event data: 9999987484016651
pallet: creditcoin event name: AddressRegistered
event types H256 event data: 0x8019f217d727e9c170f0a6664a315ea7ac16eee371c84d2080d4ab951ad6806e
pallet: creditcoin event name: AddressRegistered
event types {"blockchain":"PalletCreditcoinBlockchain","value":"Bytes","owner":"AccountId32"} event data: {"blockchain":{"ethereum":null},"value":"0x3c6a6762f969b36bb1a6dbd598a5dc9800284d77","owner":"5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"}
pallet: system event name: ExtrinsicSuccess
event types {"weight":"u64","class":"FrameSupportWeightsDispatchClass","paysFee":"FrameSupportWeightsPays"} event data: {"weight":125010000,"class":"Normal","paysFee":"Yes"}
```

however the process still doesn't exit!